modin/numpy/test/test_array.py

```diff
@@ -0,0 +1,778 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import numpy
+import pytest
+import warnings
+import modin.numpy as np
+
+
+@pytest.mark.parametrize("size", [100, (2, 100), (100, 2), (1, 100), (100, 1)])
+def test_repr(size):
+    numpy_arr = numpy.random.randint(-100, 100, size=size)
+    modin_arr = np.array(numpy_arr)
+    assert repr(modin_arr) == repr(numpy_arr)
+
+
+@pytest.mark.parametrize("size", [100, (2, 100), (100, 2), (1, 100), (100, 1)])
+def test_shape(size):
+    numpy_arr = numpy.random.randint(-100, 100, size=size)
+    modin_arr = np.array(numpy_arr)
+    assert modin_arr.shape == numpy_arr.shape
+
+
+def test_dtype():
+    numpy_arr = numpy.array([[1, "2"], [3, "4"]])
+    modin_arr = np.array([[1, "2"], [3, "4"]])
+    assert modin_arr.dtype == numpy_arr.dtype
+    modin_arr = modin_arr == modin_arr.T
+    numpy_arr = numpy_arr == numpy_arr.T
+    assert modin_arr.dtype == numpy_arr.dtype
+
+
+@pytest.mark.parametrize("operand1shape", [100, (3, 100)])
+@pytest.mark.parametrize("operand2shape", [100, (3, 100)])
+@pytest.mark.parametrize(
+    "operator",
+    [
+        "__add__",
+        "__sub__",
+        "__truediv__",
+        "__mul__",
+        "__rtruediv__",
+        "__rmul__",
+        "__radd__",
+        "__rsub__",
+        "__ge__",
+        "__gt__",
+        "__lt__",
+        "__le__",
+        "__eq__",
+        "__ne__",
+    ],
+)
+def test_basic_arithmetic_with_broadcast(operand1shape, operand2shape, operator):
+    """Test of operators that support broadcasting."""
+    operand1 = numpy.random.randint(-100, 100, size=operand1shape)
+    operand2 = numpy.random.randint(-100, 100, size=operand2shape)
+    modin_result = getattr(np.array(operand1), operator)(np.array(operand2))
+    numpy_result = getattr(operand1, operator)(operand2)
+    if operator not in ["__truediv__", "__rtruediv__"]:
+        numpy.testing.assert_array_equal(
+            modin_result._to_numpy(),
```

Comment:
We should probably add a high-level `to_numpy` method similar `to_pandas` we have.

Reply:
I'm not sure I follow? The DataFrame and Series classes have a `_to_pandas()` method, so I copied that style and gave modin numpy arrays a `_to_numpy()` method.

Reply:
I mean shouldn't we add `modin.utils.to_numpy` method like we have `modin.utils.to_pandas`?

Reply:
Good catch - will add!